### PR TITLE
fix: migration 2017_01_28_222114_remove_viewed_at_from_contacts

### PR DESCRIPTION
**Pull request by JensGryspeert**
Issue #6960

### CODE REVIEW

File: database/migrations/2017_01_28_222114_remove_viewed_at_from_contacts.php [hunk 1]
```diff
@@ -14,6 +14,10 @@ class RemoveViewedAtFromContacts extends Migration
     public function up()
     {
         Schema::table('contacts', function (Blueprint $table) {
+            if (!$table->hasColumn('viewed_at')) {
+                return;
+            }
+
             $table->dropColumn(
                 'viewed_at'
             );
```
Comment (asbiin) [inline database/migrations/2017_01_28_222114_remove_viewed_at_from_contacts.php:23]: If prefer to write it:
```suggestion
            if ($table->hasColumn('viewed_at')) {
                $table->dropColumn('viewed_at');
            }
```

File: database/migrations/2017_01_28_222114_remove_viewed_at_from_contacts.php [hunk 2]
```diff
@@ -28,6 +32,10 @@ public function up()
     public function down()
     {
         Schema::table('contacts', function (Blueprint $table) {
+            if ($table->hasColumn('viewed_at')) {
+                return;
+            }
+
             $table->dateTime('viewed_at')->nullable();
```
Comment (asbiin) [inline database/migrations/2017_01_28_222114_remove_viewed_at_from_contacts.php:39]:
```suggestion
            if (!$table->hasColumn('viewed_at')) {
                $table->dateTime('viewed_at')->nullable();
            }
```

**Review by asbiin — CHANGES_REQUESTED** [review]
Thanks. See my comment

**Comment by JensGryspeert** [comment]
> Thanks. See my comment

I squashed the modifications. The reason I work with early returns is to avoid if nesting in bigger code pages - to used at it haha sorry.

**Review by asbiin — APPROVED** [review]

File: database/migrations/2017_01_28_222114_remove_viewed_at_from_contacts.php [hunk 3]
```diff
@@ -14,9 +14,9 @@ class RemoveViewedAtFromContacts extends Migration
     public function up()
     {
         Schema::table('contacts', function (Blueprint $table) {
-            $table->dropColumn(
-                'viewed_at'
-            );
+            if ($table->hasColumn('viewed_at')) {
+                $table->dropColumn('viewed_at');
+            }
         });
```
Comment (asbiin) [inline database/migrations/2017_01_28_222114_remove_viewed_at_from_contacts.php:20]: Actually `Blueprint::hasColumn` does not exist, you have to write it:
```suggestion
        if (Schema::hasColumn('contacts', 'viewed_at')) {
            Schema::table('contacts', function (Blueprint $table) {
                $table->dropColumn('viewed_at');
            });
        }
```

File: database/migrations/2017_01_28_222114_remove_viewed_at_from_contacts.php [hunk 4]
```diff
@@ -28,7 +28,9 @@ public function up()
     public function down()
     {
         Schema::table('contacts', function (Blueprint $table) {
-            $table->dateTime('viewed_at')->nullable();
+            if ($table->hasColumn('viewed_at')) {
+                $table->dateTime('viewed_at')->nullable();
+            }
         });
```
Comment (asbiin) [inline database/migrations/2017_01_28_222114_remove_viewed_at_from_contacts.php:34]: same:
```suggestion
        if (! Schema::hasColumn('contacts', 'viewed_at')) {
            Schema::table('contacts', function (Blueprint $table) {
                $table->dateTime('viewed_at')->nullable();
            });
        }
```

**Review by asbiin — CHANGES_REQUESTED** [review]
I've missed this issue the first time ->

**Pull request merged**